### PR TITLE
Log mirror path

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

We have the option to use a mirror and mount repositories local to Uyuni Server.
But it is not easy to find out where Uyuni search for a repository on which local path.

This PR add now log messages (info level) which print the possible paths to the logs when this class
is turned into logging also info messages. 
They also show alternative paths which are possible for some repositories.

## GUI diff

No difference.

Log Example:

```
2023-09-04 17:11:26,156 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - SCC mirrorpath: /mirror/SUSE/Products/SLE-Module-Server-Applications/15-SP4/x86_64/product
2023-09-04 17:11:26,160 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - SMT mirrorpath for 'ubuntu-2204-amd64-main': /mirror/repo/RPMMD/ubuntu-2204-amd64-main
2023-09-04 17:11:26,160 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - Common mirrorpath for 'ubuntu-2204-amd64-main': /mirror/ubuntu/dists/jammy/main/binary-amd64
2023-09-04 17:11:26,160 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - Default mirrorpath for 'ubuntu-2204-amd64-main': /mirror/archive.ubuntu.com/ubuntu/dists/jammy/main/binary-amd64
2023-09-04 17:11:26,161 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - SMT mirrorpath for 'ubuntu-2204-amd64-main-updates': /mirror/repo/RPMMD/ubuntu-2204-amd64-main-updates
2023-09-04 17:11:26,161 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - Common mirrorpath for 'ubuntu-2204-amd64-main-updates': /mirror/ubuntu/dists/jammy-updates/main/binary-amd64
2023-09-04 17:11:26,162 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-8] INFO  com.redhat.rhn.manager.content.MgrSyncUtils - Default mirrorpath for 'ubuntu-2204-amd64-main-updates': /mirror/archive.ubuntu.com/ubuntu/dists/jammy-updates/main/binary-amd64
```

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: only add log messages for QE

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/20637
Tracks https://github.com/SUSE/spacewalk/pull/22452

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java [hunk 1]
```diff
@@ -251,28 +255,37 @@ public static URI urlToFSPath(String urlString, String name) {
...
+            LOG.info("SCC mirrorpath: {}", mirrorPath);
         }
         else if (name != null) {
             // Case 3
             // everything after the first space are suffixes added to make things unique
-            String[] parts  = URLDecoder.decode(name, StandardCharsets.UTF_8).split("[\\s//]");
+            String[] parts  = URLDecoder.decode(name, StandardCharsets.UTF_8).split("[\\s/]");
             if (!(parts[0].isBlank() || parts[0].equals(".."))) {
                 File oldMirrorPath = Paths.get(dataPath.getAbsolutePath(), "repo", "RPMMD", parts[0]).toFile();
+                LOG.info("SMT mirrorpath for '{}': {}", name, oldMirrorPath);
```
Comment (lucidd) [inline java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java:266]: Is this log statement at the right spot or was it supposed to be in the if statement?

File: java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java [hunk 2]
```diff
@@ -251,28 +255,37 @@ public static URI urlToFSPath(String urlString, String name) {
...
                 if (oldMirrorPath.exists()) {
                     mirrorPath = oldMirrorPath;
                 }
                 else {
                     // mirror in a common folder (bsc#1201753)
                     File commonMirrorPath = Paths.get(dataPath.getAbsolutePath(), path).toFile();
+                    LOG.info("Common mirrorpath for '{}': {}", name, commonMirrorPath);
```
Comment (lucidd) [inline java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java:273]: same question as above

**Review by lucidd — APPROVED** [review]

**Review by mikeletux — APPROVED** [review]
LGTM

File: java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java (hunk 1 above)
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java:266]: This is the right place. We want to show the "options" where we look on the mirror. Users might not know where to put the stuff. So it does not exist. Putting it in the if would only show it when they found the right location.

File: java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java (hunk 2 above)
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java:273]: same as above

**Pull request merged**